src/citrine/resources/file_link.py

```diff
@@ -289,3 +297,30 @@ def _complete_upload(self, dest_name: str, uploader: _Uploader):
 
         url = self._get_path(file_id) + '/versions/{}'.format(version)
         return FileLink(filename=dest_name, url=url)
+
+    def download(self, file_link: FileLink, local_path: str):
+        """
+        Download the file associated with a given FileLink to the local computer.
+
+        Parameters
+        ----------
+        file_link: FileLink
+            Resource referencing the external file.
+        local_path: str
+            Path to save file on the local computer. If `local_path` is a directory,
+            then the filename of this FileLink object will be appended to the path.
+
+        """
+        directory, filename = os.path.split(local_path)
+        if not filename:
+            filename = file_link.filename
+        if not os.path.isdir(directory):
+            os.mkdir(directory)
+        local_path = os.path.join(directory, filename)
```

Comment:
reusing the same variable name in local scope is confusing. Though maybe this is normal in python? Not sure, but feels funky to read.

Reply:
`local_path` isn't supposed to change in general. I'm just handling the case where the user supplies a directory instead of a full path, and I'm constructing the full path for them. For that reason it felt appropriate to use the same variable name. 🤷‍♂

Reply:
But the file will be downloaded to the absolute path, so I should add checks to make sure that the user provided a legitimate absolute path.

Reply:
yeah that's no problem. Just looked weird, but sounds like reasonable naming